### ISSUE TITLE
Comment was added with date unknown

### ISSUE THREAD

**Issue by trang** [Tatoeba/tatoeba2#1948]
I received an email notification for a comment that has been saved without a date. 
Note: the date and time of the comment should be around 2019-08-30 15:12:00 (UTC).

https://tatoeba.org/eng/sentences/show/8135701#comment-1121167

![Screenshot_2019-08-30 English example sentence To those who don't know him, Mennad Bakir is a sweet and charming man - Tato](https://user-images.githubusercontent.com/221850/64036657-fda9d280-cb53-11e9-9998-1d70c33d2e28.png)

**Comment by AndiPersti** [comment]
Other comments in the database with an unknown date:
https://tatoeba.org/eng/sentences/show/4012427#comment-1106349
https://tatoeba.org/eng/sentences/show/8055150#comment-1138486
https://tatoeba.org/eng/sentences/show/686252#comment-1136398

The problem exists for all UI languages which don't use arabic numbers (currently [Bangla](https://dev.tatoeba.org/wall/show_message/31684#message_31684), [Marathi](https://dev.tatoeba.org/wall/show_message/31683#message_31683) and [Arabic](https://dev.tatoeba.org/wall/show_message/31681#message_31681)) and at least for adding sentence comments and wall messages.

#1881 may be related.

**Comment by AndiPersti** [comment]
@Trang I'm afraid we have a bigger problem. If you use the Arabic UI while adding sentences the date also isn't registered. Here's an excerpt from the current database dump (`sentences_detailed.csv`):

    $ grep '0000-00-00' sentences_detailed.csv | tail
    8167498 ara     دخلت على زوجتي فوجدتها وحيدة.   Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167501 ara     إنه المطر!      Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167506 ara     هطل المطر.      Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167508 ara     رآيت تلك الفتاة؟ وجهها يلمع.    Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167517 ber     Uckiɣ-nn / usiɣ-nn.     Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167522 ber     tanemmirt-nnek umi d-tuwted abrid-a akkʷ ḥma ad di temsafaḍed.  Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167526 ber     Ur ẓdaṛeɣ ad ssuduɣ ayyis-a. / Ur zemireɣ ad niyeɣ ayyis-a.     Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167529 ber     Laḥ ijwayen-inu.        Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167533 ber     Ili aẓẓiḍeṛ d terrugza. Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00
    8167546 ber     Annaz-nnek ar ittṛeẓẓa aẓṛu.    Tariqidufqqir   0000-00-00 00:00:00     0000-00-00 00:00:00

All this sentences are a few weeks old. I'm pretty sure user Tariqidufqqir is using the Arabic UI (I've sent a PM for confirmation).
(We currently don't display the creation date anywhere so that problem was hidden until now. The dates in the contributions logs come from the `contributions` table which stores the correct date.)

Dumping the entity of a new sentence gives me for example:

    2019-12-29 19:24:01 Debug: ar
    2019-12-29 19:24:01 Debug: {
        "text": "another arabic",
        "user_id": 4,
        "correctness": 0,
        "hash": "2oa64m2\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
        "license": "CC BY 2.0 FR",
        "based_on_id": 0,
        "lang": "fra",
        "created": "٢٠١٩-١٢-٢٩ ١٩:٢٤:٠١",
        "modified": "٢٠١٩-١٢-٢٩ ١٩:٢٤:٠١",
        "script": null,
        "id": 24
    }
It looks like, the `Timestamp` behavior uses the current locale when creating the datetime string and for some reason the numbers in that string are formatted in that locale.

**Comment by AndiPersti** [comment]
I've found the problem: 
https://github.com/cakephp/cakephp/blob/69643f58dd110a4c5cce7ff867707de176c64bc4/src/ORM/Behavior/TimestampBehavior.php#L149
The returned value by `new Time()` is influenced by the current locale:

    2019-12-29 20:10:28 Debug: en
    2019-12-29 20:10:28 Debug: 2019-12-29 20:10:28
    2019-12-29 20:10:49 Debug: ar
    2019-12-29 20:10:49 Debug: ٢٠١٩-١٢-٢٩ ٢٠:١٠:٤٩
    2019-12-29 20:11:06 Debug: bn
    2019-12-29 20:11:06 Debug: ২০১৯-১২-২৯ ২০:১১:০৬

**Comment by AndiPersti** [comment]
Actually the real issue lies here:
https://github.com/cakephp/cakephp/blob/69643f58dd110a4c5cce7ff867707de176c64bc4/src/ORM/Behavior/TimestampBehavior.php#L208-L216

We set the column type to `string` for most datetime columns. Thus the timestamp (a `Cake\I18n\Time` object) will get converted to a string (with locale dependent numerals).

What's the reason for setting the column type to `string`? I've already found that a little bit strange while working on internationalizing dates. (And this will break whenever we upgrade CakePHP as the deprecation warning mentions.)

If I leave the `created` column as is (i.e. the default `datetime` type), the date will be stored correctly for all locales.
